### PR TITLE
[ISSUE #2917] Shenyu add logging-kafka plugin

### PR DESCRIPTION
**Pull request by qifanyyy**
- In this pull request, the updated license includes Kafka dependency.
- The Kafka logging plugin is implemented
- shenyu-spring-boot-starter-plugin-logging-kafka is implemented
- ShenyuThreadPoolExecutor is updated in AbstractLogCollector.java

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by qicz** [comment]
@qifanyyy pls check style
`Error:  Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (validate) on project shenyu-common: You have 1 Checkstyle violation. -> [Help 1]`

**Comment by qifanyyy** [comment]
Yes, tried to resolve conflicts with other pull request and the style is not matched when resolving conflicts, it has been resolved.

>

**Comment by JooKS-me** [comment]
@qifanyyy Pls merge master to resolve integration-test error, see #3611

**Review by tuohai666 — APPROVED** [review]

File: shenyu-common/src/main/java/org/apache/shenyu/common/enums/PluginEnum.java [hunk 1]
```diff
@@ -137,6 +137,11 @@ public enum PluginEnum {
      */
     LOGGING_ROCKETMQ(170, 0, "loggingRocketMQ"),
 
+    /**
+     * Logging RocketMQ plugin enum.
```
[the remark this answers is not in the document]
Comment (qifanyyy) [reply shenyu-common/src/main/java/org/apache/shenyu/common/enums/PluginEnum.java:141]: Noted. Will be updated in the next pull request.

**Pull request merged**